app/leave-app/backend/modules/authorization/types.bal

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2024 WSO2 LLC. (http://www.wso2.org).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Application specific role mapping.
+public type AppRoles record {|
+    # Role for the recruitment team
+    string recruitmentTeamRole;
+    # Role for the head of people operations 
+    string headPeopleOperationsRole;
+|};
+
+# User info custom type for Asgardeo token.
+public type CustomJwtPayload record {
```

Comment:
Why do we have to keep this as an open record here?

Reply:
The `CustomJwtPayload` is kept as an open record because `jwt:Payload` contains additional fields that are not part of our `CustomJwtPayload` type. When `CustomJwtPayload` is closed, these extra fields lead to a ConversionError during type casting. Keeping it open ensures that cloneWithType can handle fields present in `jwt:Payload` without conversion issues.

```
[jwt:Header, jwt:Payload]|jwt:Error result = jwt:decode(key);
    if result is jwt:Error {
        return result;
    }
    CustomJwtPayload|error userInfo = result[1].cloneWithType();
```

Reply:
Pl check the possiblity of using,

```
public type CustomJwtPayload record {|
    # User email 
    string email;
    # User groups
    string[] groups;
    json...;
|};
```

Reply:
@suhand I had a chat with ballerina team members and they said it's okay to go with this approach. 
Note: `json` type is similar to `anydata` type with the inclusion of `regex` and `table` type.